lib/charms/layer/bigtop_spark.py

```diff
@@ -66,40 +62,69 @@ def install_benchmark(self):
             unitdata.kv().flush(True)
 
     def setup(self):
-        self.set_log_permissions()
+        self.dist_config.add_users()
+        self.dist_config.add_dirs()
         self.install_demo()
         self.open_ports()
 
+    def setup_hdfs_logs(self):
+        # create hdfs storage space for history server
+        dc = self.dist_config
+        events_dir = dc.path('spark_events')
+        events_dir = 'hdfs://{}'.format(events_dir)
+        utils.run_as('hdfs', 'hdfs', 'dfs', '-mkdir', '-p', events_dir)
+        utils.run_as('hdfs', 'hdfs', 'dfs', '-chown', '-R', 'ubuntu:spark',
+                     events_dir)
+        return events_dir
+
     def configure(self, available_hosts):
+        # This is the core logic of setting up spark.
+        # Two flags are needed: 
+        # 1) Namenode exists aka HDFS is there
+        # 2) Resource manager exists aka YARN is ready
```

Comment:
Would prefer to see a standard style of function docstring here:

    """
    This is the core logic ...

    """

Ignore if we are deliberately using a different style in this module.